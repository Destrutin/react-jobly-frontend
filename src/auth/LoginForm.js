import React, {useState} from "react";
import {useNavigate} from "react-router-dom";

function LoginForm({login}) {
    const navigate = useNavigate();
    const [formData, setFormData] = useState({
        username: "",
        password: ""
    });

    const handleChange = e => {
        const {name, value} = e.target;
        setFormData(data => ({...data, [name]:value}));
    }
    
    const handleSubmit = async (e) => {
        e.preventDefault();
        try {
            await login(formData);
            navigate("/companies");
        } catch (err) {
            console.error(err);
        }
    }

    return (
        <div className="LoginForm">
            <h1>Log In</h1>
            <div className="login-card">
                <form onSubmit={handleSubmit}>
                    <div>
                        <label>Username</label>
                        <input
                            name="username"
                            value={formData.username}
                            onChange={handleChange}
                            required
                        />
                    </div>
                    <div>
                        <label>Password</label>
                        <input
                            name="password"
                            value={formData.password}
                            onChange={handleChange}
                            required
                        />
                    </div>

                    <button
                        className="login-button"
                        onSubmit={handleSubmit}
                    >Submit</button>
                </form>
            </div>
        </div>
    )
}

export default LoginForm;