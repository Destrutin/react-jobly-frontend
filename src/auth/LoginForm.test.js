import React from 'react';
import {render, fireEvent} from "@testing-library/react";
import {MemoryRouter} from 'react-router-dom';
import LoginForm from './LoginForm';
import { act } from 'react-dom/test-utils';

test('renders login form without crashing', () => {
    render(
        <MemoryRouter>
            <LoginForm/>
        </MemoryRouter>
    )
})

test('matches snapshot', () => {
    const testLogin = [{username: "Fang", password: "Fangiscool"}]
    const {asFragment} = render(
        <MemoryRouter>
            <LoginForm login={testLogin}/>
        </MemoryRouter>
    )
    expect(asFragment()).toMatchSnapshot();
})

test('Login form submits with correct data', async () => {
    const mockLogin = jest.fn();

    const {getByLabelText, getByText} = render(
        <MemoryRouter>
            <LoginForm login={mockLogin}></LoginForm>
        </MemoryRouter>
    )
    
    await act(async () => {
        fireEvent.change(getByLabelText('Username'), {target: {value: 'Fang'}});
    fireEvent.change(getByLabelText('Password'), {target: {value: 'Fangiscool'}});

    fireEvent.submit(getByText('Submit'));

    expect(mockLogin).toHaveBeenCalled();
    })
})