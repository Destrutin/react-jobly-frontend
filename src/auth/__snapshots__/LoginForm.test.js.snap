// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="LoginForm"
  >
    <h1>
      Log In
    </h1>
    <div
      class="login-card"
    >
      <form>
        <div>
          <label
            for="username"
          >
            Username
          </label>
          <input
            id="username"
            name="username"
            required=""
            value=""
          />
        </div>
        <div>
          <label
            for="password"
          >
            Password
          </label>
          <input
            id="password"
            name="password"
            required=""
            value=""
          />
        </div>
        <button
          class="login-button"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
